Pyrado/scripts/training/qq-su_pddr.py

```diff
@@ -0,0 +1,160 @@
+from pyrado.policies.feed_forward.fnn import FNNPolicy
+from pyrado.policies.special.environment_specific import QQubeSwingUpAndBalanceCtrl
+from pyrado.spaces import ValueFunctionSpace
+from pyrado.utils.argparser import get_argparser
+from pyrado.utils.data_types import EnvSpec
+
+
+if __name__ == "__main__":
+    parser = get_argparser()
+    parser.add_argument("--freq", type=int, default=250)
+    parser.add_argument("--max_iter_teacher", type=int, default=200)
+    parser.add_argument("--train_teachers", action="store_true", default=False)
+    parser.add_argument("--num_teachers", type=int, default=2)
+    parser.add_argument("--max_iter", type=int, default=500)
+    parser.add_argument("--num_epochs", type=int, default=10)
+
+    # Parse command line arguments
+    args = parser.parse_args()
+
+    # Set seed if desired
+    pyrado.set_seed(args.seed, verbose=True)
+    use_cuda = args.device == "cuda"
+    descr = f"_{args.max_steps}st_{args.freq}Hz"
+
+    # Environment
+    env_hparams = dict(dt=1 / args.freq, max_steps=args.max_steps)
+    env_real = QQubeSwingUpSim(**env_hparams)
+    ex_dir = setup_experiment(QQubeSwingUpSim.name, f"{PDDR.name}_{QQubeSwingUpAndBalanceCtrl.name}{descr}")
+
+    if args.train_teachers:
+        # Policy
+        teacher_policy_hparam = dict(
+            hidden_sizes=[64, 64], hidden_nonlin=to.relu, output_nonlin=to.tanh, use_cuda=use_cuda
+        )
+        teacher_policy = FNNPolicy(spec=env_real.spec, **teacher_policy_hparam)
+
+        # Reduce weights of last layer, recommended by paper
+        for p in teacher_policy.net.output_layer.parameters():
+            with to.no_grad():
+                p /= 100
+
+        # Critic
+        vfcn_hparam = dict(hidden_sizes=[32, 32], hidden_nonlin=to.relu)
+        vfcn = FNNPolicy(spec=EnvSpec(env_real.obs_space, ValueFunctionSpace), **vfcn_hparam)
+        critic_hparam = dict(
+            gamma=0.9844224855479998,
+            lamda=0.9700148505302241,
+            num_epoch=5,
+            batch_size=500,
+            standardize_adv=False,
+            lr=7.058326426522811e-4,
+            max_grad_norm=6.0,
+            lr_scheduler=lr_scheduler.ExponentialLR,
+            lr_scheduler_hparam=dict(gamma=0.999),
+        )
+        critic = GAE(vfcn, **critic_hparam)
+
+        # Subroutine
+        teacher_algo_hparam = dict(
+            eps_clip=0.12648736789309026,
+            min_steps=env_real.max_steps,
+            num_epoch=7,
+            batch_size=500,
+            std_init=0.7573286998997557,
+            lr=6.999956625305722e-04,
+            max_grad_norm=1.0,
+            num_workers=8,
+            lr_scheduler=lr_scheduler.ExponentialLR,
+            lr_scheduler_hparam=dict(gamma=0.999),
+            max_iter=args.max_iter_teacher,
+            critic=critic,
+        )
+
+        teacher_algo = PPO
+
+    else:
+        teacher_policy = None
+        teacher_algo = None
+        teacher_algo_hparam = None
```

Comment:
Is that part still necessary, now that you inherit from InterruptableAlgorithm? I mean if you do not want to train the teachers, you could/should copy existing teachers, right?